### PR TITLE
feat(examples): add example on how to use flux as a library

### PR DESCRIPTION
**Pull request by fntlnz**
Signed-off-by: Lorenzo Fontana <lo@linux.com>

### CODE REVIEW

**Comment by fntlnz** [comment]
@nathanielc @affo PTAL

File: examples/library/main.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		}
+		spec, err := flux.ToSpec([]values.Value{t}, nowTime.Time().Time())
+		if err != nil {
+			panic(err)
+		}
+		compiler := lang.SpecCompiler{
+			Spec: spec,
+		}
+
+		querier := NewQuerier()
```
Comment (affo) [inline examples/library/main.go]: I think you can use:

```
import "github.com/influxdata/flux/cmd/flux/cmd"

cmd.NewQuerier()
```

Comment (fntlnz) [reply examples/library/main.go]: thanks for the suggestion @affo - changed that

**Comment by fntlnz** [comment]
good call @nathanielc - please take another look now

**Comment by andig** [comment]
Great pr for an tutorial. It would be nice to add an example on how to pass data into the flux library for processing, too.

**Comment by andig** [comment]
@fntlnz too bad 😢

**Comment by affo** [comment]
This is a great example and we have only lost track of it.  

@fntlnz If now it requires some changes to work out, I can perform them on this branch.

@nathanielc could you please review this?

**Comment by andig** [comment]
> It would be nice to add an example on how to pass data into the flux library for processing, too.

@affo would that be something you could add on top?

**Comment by affo** [comment]
@andig sure.

But what do you mean with passing data?

```
import "csv"

data = "
#datatype,string,long,long,string
#group,false,false,false,true
#default,_result,,,
,result,table,value,tag
,,0,10,a
,,0,10,a
,,1,20,b
,,1,20,b
,,2,30,c
,,2,30,c
,,3,40,d
,,3,40,d
"

csv.from(csv: data) |> yield(name: "res")
```

This is a valid flux script.

Do you mean encoding a table from go by using an `Encoder`? Yep, I could do that.

Anyway, this PR is about this example. @andig if you want to see another example, can you please file an issue for that in Flux? Feel free to tag me :)

**Comment by andig** [comment]
> But what do you mean with passing data?

@affo you example is already enough to reverse anything else that might be needed by looking at the `csv.from()` function. Much appreciated, thank you!

**Comment by affo** [comment]
@andig the PR got merged to master 🥂 

That folder is there to be enhanced with other examples. If you want to see other examples, feel free to open issues in Flux and tag me in!

If you want to contribute with yours too, you are welcome 😎 😄

**Pull request merged**